### ISSUE TITLE
`prefer-string-replace-all` may suggest an incorrect alternative

### ISSUE THREAD

**Issue by Jiralite** [sindresorhus/eslint-plugin-unicorn#1961]
`unicorn/prefer-string-replace-all` seems to be reporting incorrect alternatives for regular expressions with the global flag & with a `|`. Consider the following:

```TypeScript
declare const content: string;

content.replace(/\*|_/g, ""); // Prefer `String#replaceAll()` over `String#replace()`. eslint(unicorn/prefer-string-replace-all)
```

Upon applying the suggested fix, the code results to:

```TypeScript
declare const content: string;

content.replaceAll('*|_', "");
```

The regular expression has been lost and the code obviously does not have the original intentions.

It looks like the issue is that `|` is not included here:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/505a2039627350fe7ae0b990d48cd035d8184949/rules/prefer-string-replace-all.js#L20-L23

<hr>

- Node.js: 18.8.0
- @typescript-eslint/eslint-plugin: 5.42.1
- @typescript-eslint/parser: 5.42.1
- eslint: 8.27.0
- eslint-plugin-unicorn: 44.0.2
- typescript 4.8.4
